controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -252,14 +256,72 @@ func (r *OCSInitializationReconciler) ensureRookCephOperatorConfigExists(initial
 			Namespace: initialData.Namespace,
 		},
 	}
-	err := r.Client.Create(r.ctx, rookCephOperatorConfig)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		r.Log.Error(err, fmt.Sprintf("Failed to create %s configmap", util.RookCephOperatorConfigName))
+
+	opResult, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rookCephOperatorConfig, func() error {
+
+		csiPluginDefaults := defaults.DaemonPlacements[defaults.CsiPluginKey]
+		csiProvisionerDefaults := defaults.DaemonPlacements[defaults.CsiProvisionerKey]
+
+		csiPluginTolerations := r.getCsiTolerations(defaults.CsiPluginKey)
+		csiProvisionerTolerations := r.getCsiTolerations(defaults.CsiProvisionerKey)
+
+		if rookCephOperatorConfig.Data == nil {
+			rookCephOperatorConfig.Data = make(map[string]string)
+		}
+
+		updateTolerationsConfigFunc := func(tolerations, defaults []corev1.Toleration, configMapKey string, modifiedFlag *bool) error {
+
+			if tolerations != nil {
+				updatedTolerations := append(tolerations, defaults...)
+				tolerationsYAML, err := yaml.Marshal(updatedTolerations)
+				if err != nil {
+					return err
+				}
+				rookCephOperatorConfig.Data[configMapKey] = string(tolerationsYAML)
+				*modifiedFlag = true
+			} else if tolerations == nil && *modifiedFlag {
+				delete(rookCephOperatorConfig.Data, configMapKey)
+				*modifiedFlag = false
+			}
+
+			return nil
+		}
+
+		if err := updateTolerationsConfigFunc(
+			csiPluginTolerations, csiPluginDefaults.Tolerations, "CSI_PLUGIN_TOLERATIONS",
+			&initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified); err != nil {
+			return err
+		}
+
+		err := updateTolerationsConfigFunc(
+			csiProvisionerTolerations, csiProvisionerDefaults.Tolerations, "CSI_PROVISIONER_TOLERATIONS",
+			&initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified)
+
+		return err
+	})
+
+	if err != nil {
+		r.Log.Error(err, "Failed to create/update rook-ceph-operator-config configmap")
 		return err
 	}
+	r.Log.Info("Successfully created/updated rook-ceph-operator-config configmap", "OperationResult", opResult)
+
 	return nil
 }
 
+func (r *OCSInitializationReconciler) getCsiTolerations(csiTolerationKey string) []corev1.Toleration {
+
+	var tolerations []corev1.Toleration
+
+	for _, sc := range r.clusters.GetStorageClusters() {
```

Comment:
This for loop will create an in-memory copy of the Storage Cluster objects in r.clusters (the slice is not a pointers slice). 
Can you please change to: 
```suggestion
	clusters :=  r.clusters.GetStorageClusters()
	for i := range clusters {
		sc := &clusters[i]
```

Reply:
done